sdks/python/apache_beam/runners/dataflow/internal/dependency.py

```diff
@@ -453,36 +457,85 @@ def _build_setup_package(setup_file, temp_dir, build_setup_args=None):
     os.chdir(saved_current_directory)
 
 
-def _stage_beam_sdk_tarball(sdk_remote_location, staged_path, temp_dir):
-  """Stage a Beam SDK tarball with the appropriate version.
+def _desired_sdk_filename_in_staging_location(sdk_location):
+  """Returns the name that SDK file should have file in the staging location.
 
   Args:
-    sdk_remote_location: A GCS path to a SDK tarball or a URL from
-      the file can be downloaded.
-    staged_path: GCS path where the found SDK tarball should be copied.
+    sdk_location: Full path to SDK file.
+  """
+  if sdk_location.endswith('.whl'):
+    if sdk_location.startswith('http'):
+      raise RuntimeError('Staging SDK wheel from an HTTP location is currently'
+                         'not supported.')
+    _, wheel_filename = FileSystems.split(sdk_location)
+    if (wheel_filename.startswith('apache_beam') or
+        wheel_filename.startswith('google_cloud_dataflow')):
+      return wheel_filename
+    else:
+      raise RuntimeError('Unrecognized SDK wheel file: %s' % sdk_location)
+  else:
+    return names.DATAFLOW_SDK_TARBALL_FILE
+
+
+def _stage_beam_sdk(sdk_remote_location, staging_location, temp_dir):
+  """Stages a Beam SDK file with the appropriate version.
+
+  Args:
+    sdk_remote_location: A GCS path to a SDK file or a URL from which
+      the file can be downloaded. The SDK file can be a tarball or a wheel.
+      Set to 'pypi' to download and stage a wheel and source SDK from PyPi.
+    staging_location: A GCS bucket where the SDK file should be copied.
     temp_dir: path to temporary location where the file should be downloaded.
 
+  Returns:
+    A list of SDK files that were staged to the staging location.
+
   Raises:
-    RuntimeError: If wget on the URL specified returs errors or the file
-      cannot be copied from/to GCS.
+    RuntimeError: if staging was not successful.
   """
   if (sdk_remote_location.startswith('http://') or
       sdk_remote_location.startswith('https://')):
-    logging.info(
-        'Staging Beam SDK tarball from %s to %s',
-        sdk_remote_location, staged_path)
     local_download_file = _dependency_file_download(
         sdk_remote_location, temp_dir)
+    staged_name = _desired_sdk_filename_in_staging_location(local_download_file)
```

Comment:
Would not this support staging a wheel file from an http location, since we just downloaded it above as a local file?

Reply:
As I said above, as long as we preserve the original wheel name, it will work.